### PR TITLE
inject and simplify

### PR DESCRIPTION
**Pull request by kkraune**

### CODE REVIEW

File: vespa-cloud/album-recommendation-docproc/src/main/java/ai/vespa/example/album/LyricsDocumentProcessor.java [hunk 1]
```diff
@@ -110,10 +114,4 @@ private void setLyrics(Document document, Document lyricsDoc) {
         }
     }
 
-    @Override
-    public void deconstruct() {
-        super.deconstruct();
-        asyncSession.destroy();
```
Comment (jonmv) [inline vespa-cloud/album-recommendation-docproc/src/main/java/ai/vespa/example/album/LyricsDocumentProcessor.java:116]: Should still destroy this on shutdown.

**Review by jonmv — CHANGES_REQUESTED** [review]
Also do destroy the `AsyncSession` in the doc proc :)

File: vespa-cloud/album-recommendation-docproc/src/main/java/ai/vespa/example/album/LyricsDocumentProcessor.java (hunk 1 above)
Comment (kkraune) [reply vespa-cloud/album-recommendation-docproc/src/main/java/ai/vespa/example/album/LyricsDocumentProcessor.java:116]: ok!

**Comment by kkraune** [comment]
@jonmv ptal

**Comment by jonmv** [comment]
Ah, both classes are doc-procs — the out-one still doesn't destroy its session.

**Comment by kkraune** [comment]
sorry, missed that!

**Comment by jonmv** [comment]
NO worries :)

**Pull request merged**